src/server/console/linux_virtual_terminal.cpp

```diff
@@ -305,7 +306,11 @@ int mir::LinuxVirtualTerminal::open_vt(int vt_number)
     return vt_fd;
 }
 
-boost::future<mir::Fd> mir::LinuxVirtualTerminal::acquire_device(int major, int minor)
+std::future<std::unique_ptr<mir::Device>> mir::LinuxVirtualTerminal::acquire_device(
+    int major, int minor,
+    Device::OnDeviceActivated const& on_activated,
+    Device::OnDeviceSuspended const&,
+    Device::OnDeviceRemoved const&)
```

Comment:
This tuple (of Device::OnDeviceActivated, Device::OnDeviceSuspended & Device::OnDeviceRemoved) seems to go together. Is it a struct in disguise? Or an interface in disguise?